src/core/system.js

```diff
@@ -279,16 +279,28 @@ function combinePlugins(plugins, toolbox) {
   return {}
 }
 
+function downCaseThePropsOf(obj) {
```

Comment:
nothing wrong with this... but my thoughts are leaning towards 3rd party libs for simple utilities...
eg:
```js
// https://lodash.com/docs/4.17.4#mapKeys 
import mapKeys from "lodash/mapKeys"
// ...
const lowercaseKeyComponents =  mapKeys(src.components, (val,key) => key.toLowerCase())
dest.components = {...dest.components, ...lowercaseKeyComponents }
//...
```
My reasoning here is that utilities we write are prone to error, and lack of maintenance. 
That said, this is an optional enhancement, and I'll approve the PR regardless